x/authenticator/keeper/msg_server.go

```diff
@@ -89,3 +90,38 @@ func (m msgServer) RemoveAuthenticator(goCtx context.Context, msg *types.MsgRemo
 		Success: true,
 	}, nil
 }
+
+func (m msgServer) SetActiveState(goCtx context.Context, msg *types.MsgSetActiveState) (*types.MsgSetActiveStateResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// `MsgSetActiveState` must have only one signer
+	signer := msg.GetSigners()[0]
```

Comment:
what's the risk here of having 0 signer?

Reply:
`MsgSetActiveState` can't have 0 signer as it's defined so, it grabs `MsgSetActiveState.sender` and put it there. I'm using `GetSigners` only to ensure that the signer is actually verified. If it's 0 signer, it should panic (in the test) and we know that implementation is wrong.